pkg/routeagent_driver/handlers/mtu/mtuhandler.go

```diff
@@ -123,6 +146,15 @@ func (h *mtuHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
 		}
 	}
 
+	if h.forceMss == needed {
```

Comment:
@yboaron so here we are programming both the rules right? 
i.e., existing rules to clamp tcp mss to PMTUD and
if globalnet deployment, new rules to clamp tcp mss to a fixed value. 

IMHO, I feel we should just program one of them depending on the deployment.

i.e., move this check to Init() method and in Line 105/110 if Vanilla Submariner, we will continue to use the same rules (aka PMTUD) but if its Globalnet we will program rules to clamp the MSS to a fixed value.